app/services/login/user_verifier.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'login/errors'
+
+module Login
+  class UserVerifier
+    include Accountable
+
+    attr_reader :current_user
+
+    def initialize(user)
+      @current_user = user
+    end
+
+    def perform
+      return unless current_user
+
+      user_verification_create_or_update
+    end
+
+    private
+
+    def login_type
+      @login_type ||= current_user.identity.sign_in[:service_name]
+    end
+
+    # Queries for a UserVerification on the user, based off the credential identifier
+    # If a UserVerification doesn't exist, create one and a UserAccount record associated
+    # with that UserVerification
+    def user_verification_create_or_update
+      case login_type
+      when SAML::User::MHV_MAPPED_CSID
+        find_or_create_user_verification(:mhv_uuid, current_user.mhv_correlation_id)
+      when SAML::User::IDME_CSID
+        find_or_create_user_verification(:idme_uuid, current_user.idme_uuid)
+      when SAML::User::DSLOGON_CSID
+        find_or_create_user_verification(:dslogon_uuid, current_user.identity.edipi)
+      when SAML::User::LOGINGOV_CSID
+        find_or_create_user_verification(:logingov_uuid, current_user.logingov_uuid)
+      else
+        Rails.logger.info(
+          "[Login::UserVerifier] Unknown or missing login_type for user=#{current_user.uuid}, login_type=#{login_type}"
+        )
+
+        raise Login::Errors::UnknownLoginTypeError
+      end
+    rescue => e
```

Comment:
Should we add a TODO: here since we plan on removing this?

Reply:
👍